### ISSUE TITLE
Partial dependence / model dump

### ISSUE THREAD

**Issue by avidale** [catboost/catboost#23]
Good evening! 
Is there any way to intuitively explore the model? 

E.g. in scikit-learn GB package there is the function `plot_partial_dependence` which calculates bivariate relationship between one feature and a target.
Or in XGBoost there no such built-in function, but user can parse model dump and recreate partial dependence plot or any other useful statistic from it.

What is the best way to do this in CatBoost? Will partial dependence calculation be supported? Or any way to transform the model into traversable trees?

**Comment by MichaelMonashev** [comment]
json format for dump will be useful for me too.

**Comment by annaveronika** [comment]
@kirmorozov:
To use models within search engines and other rated systems(ex: Elasticsearch, Solr)
It is good to have export capabilities to export in compatible format.
http://elasticsearch-learning-to-rank.readthedocs.io/en/latest/training-models.html#xgboost-example

References from xgboost
dmlc/xgboost#1726
dmlc/xgboost#1722

**Comment by Denisevi4** [comment]
Looking forward to it.

At least for the models with no categorical variables it should be straightforward to dump a json, right?

**Comment by avidale** [comment]
@annaveronika, if I want to implement this feature, how could I get started?

Is it sensible to reuse the CoreML transformer?

**Comment by annaveronika** [comment]
That would be great if you would help!
The CoreML doesn't support ctr-s yet, so this transformer will be not sufficient.
Here is some explanation about the code of the model:
[catboost_model_evaluation.pdf](https://github.com/catboost/catboost/files/1651062/catboost_model_evaluation.pdf)
If anything is not clear in there, please ask.

**Comment by annaveronika** [comment]
json format is implemented in version 0.10

**Comment by ser0t0nin** [comment]
@annaveronika do you have any reference to the implemented JSON-format? It is very different to the requested xgboost JSON format (and is incompatible with Elasticsearch thus), so I'd like to check, if it is possible to make a converter

**Comment by annaveronika** [comment]
We are working on the json tutorial, it'll be on github shortly. We'll post the update in https://github.com/catboost/catboost/issues/522

**Comment by ser0t0nin** [comment]
> We are working on the json tutorial, it'll be on github shortly. We'll post the update in #522

subscribed, thanks a lot

**Comment by saurzcode** [comment]
Does C/C++ API support loading the JSON dump of the model for predictions?

**Comment by annaveronika** [comment]
We have implemented json and tuturial for that. But we have not implemented plot_partial_dependence, this is a great idea for contribution, so I'm setting help wanted label.

**Comment by LyzhinIvan** [comment]
`plot_partial_dependence` for symmetric trees is available since the 0.23.2 version.
